gtsam_unstable/slam/PartialPriorFactor.h

```diff
@@ -112,13 +116,14 @@ namespace gtsam {
     Vector evaluateError(const T& p, OptionalMatrixType H) const override {
       Eigen::Matrix<double, T::dimension, T::dimension> H_local;
 
-      // If the Rot3 Cayley map is used, Rot3::LocalCoordinates will throw a runtime error
-      // when asked to compute the Jacobian matrix (see Rot3M.cpp).
-      #ifdef GTSAM_ROT3_EXPMAP
-      const Vector full_tangent = T::LocalCoordinates(p, H ? &H_local : nullptr);
-      #else
+    // If the Rot3 Cayley map is used, Rot3::LocalCoordinates will throw a runtime
+    // error when asked to compute the Jacobian matrix (see Rot3M.cpp).
+#ifdef GTSAM_ROT3_EXPMAP
```

Comment:
Are we making an assumption here on the type of T ? If so, consider template specialization